src/executor.cpp

```diff
@@ -1038,4 +1040,52 @@ void Executor::execute74()
     displayExecutionStatus(74, std::vector<types::FunctionNumber>(), true);
 }
 
-} // namespace panel
\ No newline at end of file
+types::DbusExecStatus
+    Executor::executeDbusRequests(const types::FunctionNumber funcNum)
+{
+    isThisReqFromDbus = true;
+    try
+    {
+        switch (funcNum)
+        {
+            case 21:
+            case 22:
+            case 34:
+            case 65:
+            case 66:
+            case 67:
+            case 68:
+            case 69:
+            case 70:
+                sendFuncNumToPhyp(funcNum);
+                break;
+
+            default:
+                std::runtime_error("Given function number is not supported to "
+                                   "execute via d-bus.");
+        }
+    }
+    catch (std::exception& e)
+    {
+        isThisReqFromDbus = false;
+        storeDbusExecStatus(false, "", "");
+        std::cerr << e.what() << std::endl;
+        return executionStatus;
+    }
+
+    isThisReqFromDbus = false;
+    return executionStatus;
+}
+
+void Executor::storeDbusExecStatus(bool returnState, std::string line1,
+                                   std::string line2)
+{
+    if (!isThisReqFromDbus)
+    {
+        return;
+    }
+    std::get<0>(executionStatus) = returnState;
```

Comment:
std::make_tuple(returnState, line1, line2);

just one line, will make the code look cleaner.